Description: Initialise system tray via idle #1249
 Initialise the startup of the system tray via GLib idle.
 This allows the system tray startup fully stopping black background
 and icon positioning issues.
Author: David Mohammed <fossfreedom@ubuntu.com>
Bug: https://github.com/budgie-desktop/budgie-desktop/issues/1249
Forwarded: https://github.com/budgie-desktop/budgie-desktop/pull/1366
Last-Update: 2018-03-22

--- budgie-desktop-10.4.orig/src/applets/tray/TrayApplet.vala
+++ budgie-desktop-10.4/src/applets/tray/TrayApplet.vala
@@ -39,22 +39,25 @@ public class TrayApplet : Budgie.Applet
         box.vexpand = false;
         vexpand = false;
 
-        map.connect_after(()=> {
-            maybe_integrate_tray();
-        });
+        Idle.add(()=> { 
+            map.connect_after(()=> {
+                maybe_integrate_tray();
+            });
 
 
-        show_all();
-        panel_size_changed.connect((p,i,s)=> {
-            this.icon_size = s;
-            if (tray != null) {
-                tray.set_icon_size(icon_size);
-                queue_resize();
-                tray.queue_resize();
-            }
-        });
+            show_all();
+            panel_size_changed.connect((p,i,s)=> {
+                this.icon_size = s;
+                if (tray != null) {
+                    tray.set_icon_size(icon_size);
+                    queue_resize();
+                    tray.queue_resize();
+                }
+            });
 
-        size_allocate.connect(on_size_allocate);
+            size_allocate.connect(on_size_allocate);
+            return false;
+        });
     }
 
     public override void panel_position_changed(Budgie.PanelPosition position)
